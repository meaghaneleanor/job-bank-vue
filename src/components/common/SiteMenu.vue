<template>
  <div>
    <nav role="navigation" aria-label="Primary navigation" class="ontario-site-nav">
      <button type="button" class="ontario-button ontario-site-nav__nav-toggle ontario-float-right" data-toggle="dropdown">
        <i class="ontario-icon ontario-icon--menu ontario-icon--size-medium hide-for-small-only" aria-hidden="true" /> {{ $t('header.menu') }}
      </button>

      <a v-on:click="switchLocale" class="ontario-site-nav__translation-link ontario-float-right">
        <span v-if="$i18n.locale === 'fr'" class="show-for-large">english</span>
        <span v-else class="show-for-large">français</span>
        <span class="hide-for-large">
          <abbr v-if="$i18n.locale === 'fr'" title="english">ER</abbr>
          <abbr v-else title="français">FR</abbr>
        </span>
      </a>

      <div
        id="dropdown"
        class="dropdown-pane ontario-site-nav__dropdown-pane"
        ref="dropdown"
        data-dropdown="main-menu-dropdown"
        data-v-offset="15" 
        data-position="bottom" 
        data-alignment="right"
        data-auto-focus="true" 
        data-close-on-click="true">
        <ul class="ontario-list--no-bullet">
          <li v-for="(link, index) in $t('navLinks')" :key="index" class="ontario-site-nav__dropdown-links">
            <a :href="link.href">{{ link.title }}</a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</template>

<script>
import routeTranslation from '../../translations/route-translations.js';

export default {
  methods: {
    switchLocale() {
      //get the users locale and set it to the opposite if the switch locale link is clicked
      this.$i18n.locale = (this.$i18n.locale === 'en') ? 'fr' : 'en';

      //find the opposite locale to retrieve the opposite path for routing
      let switchedLocale = (this.$i18n.locale === 'en') ? 'fr' : 'en';

      //get the full path of the current route
      let fullPath = this.$route.fullPath;

      //if a user clicks the switch locale link, retrieve the opposite locale's route path based on the current path
      this.$router.push({ path: routeTranslation[switchedLocale][fullPath]});
    }
  },
  mounted() {
    this.$foundation(this.$refs.dropdown);
  }
};
</script>
