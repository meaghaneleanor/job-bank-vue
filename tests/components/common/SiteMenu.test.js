import { shallowMount } from '@vue/test-utils';

import messages from '../../../src/translations';
import routeTranslation from '../../../src/translations/route-translations.js';
import foundation from '../../../src/plugins/vue-foundation.js';

import SiteMenu from '../../../src/components/common/SiteMenu.vue';

const $route = {
  path: '/page',
  fullPath: '/page'
};

const createComponent = shallowMount(SiteMenu, {
  mocks: {
    $t: () => {},
    $i18n: {
      locale: 'en',
      messages
    },
    $route,
    routeTranslation,
    $router: {
      push: jest.fn(),
    },
    foundation,
    switchLocale: jest.fn()
  },
});

describe('Header.vue', () => {
  describe('snapshot', () => {
    it('should display the expected html', () => {
      let wrapper = createComponent;
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('.switchLocale', () => {
      let wrapper; 

      beforeAll(() => {
        wrapper = createComponent;
        wrapper.find('.ontario-site-nav__translation-link').trigger('click');
      });

      it('should switch the locale when the language toggle is clicked', () => {
        expect(wrapper.vm.$i18n.locale).toEqual('fr');
      });

      it('should route to the correct path', () => {
        expect(wrapper.vm.$router.push).toHaveBeenCalledWith({path: '/fr/page'});
      });

      it('should switch locale again if clicked twice', () => {
        wrapper.find('.ontario-site-nav__translation-link').trigger('click');
        expect(wrapper.vm.$i18n.locale).toEqual('en');
      });
    });
  });
});
