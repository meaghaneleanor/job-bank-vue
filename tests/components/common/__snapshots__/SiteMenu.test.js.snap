// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteMenu.vue snapshot should display the expected html 1`] = `
"<div><nav role=\\"navigation\\" aria-label=\\"Primary navigation\\" class=\\"ontario-site-nav\\"><button type=\\"button\\" data-toggle=\\"dropdown\\" class=\\"ontario-button ontario-site-nav__nav-toggle ontario-float-right\\"><i aria-hidden=\\"true\\" class=\\"ontario-icon ontario-icon--menu ontario-icon--size-medium hide-for-small-only\\"></i> header.menu
    </button> <a class=\\"ontario-site-nav__translation-link ontario-float-right\\"><span class=\\"show-for-large\\">français</span> <span class=\\"hide-for-large\\"><abbr title=\\"français\\">FR</abbr></span></a> <div id=\\"dropdown\\" data-dropdown=\\"main-menu-dropdown\\" data-v-offset=\\"15\\" data-position=\\"bottom\\" data-alignment=\\"right\\" data-auto-focus=\\"true\\" data-close-on-click=\\"true\\" class=\\"dropdown-pane ontario-site-nav__dropdown-pane\\" aria-hidden=\\"true\\" data-yeti-box=\\"dropdown\\" data-resize=\\"dropdown\\"><ul class=\\"ontario-list--no-bullet\\"><li class=\\"ontario-site-nav__dropdown-links\\"><a></a></li><li class=\\"ontario-site-nav__dropdown-links\\"><a></a></li><li class=\\"ontario-site-nav__dropdown-links\\"><a></a></li><li class=\\"ontario-site-nav__dropdown-links\\"><a></a></li><li class=\\"ontario-site-nav__dropdown-links\\"><a></a></li><li class=\\"ontario-site-nav__dropdown-links\\"><a></a></li><li class=\\"ontario-site-nav__dropdown-links\\"><a></a></li><li class=\\"ontario-site-nav__dropdown-links\\"><a></a></li></ul></div></nav></div>"
`;
